### PR TITLE
Release notes for beta-20170420

### PR DESCRIPTION
**Pull request by bdarnell**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1309)

### CODE REVIEW

File: beta-20170420.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170420.windows-6.2-amd64.exe.tgz"><button id="windows" data-eventcategory="windows-binary-release-notes">Windows</button></a>
+</div>
+
+### Backwards-Incompatible Changes
+
+- Security-related command-line arguments have been changed. Instead of individual flags, certificates and keys are now loaded from `--certs-dir` (defaults to `~/.cockroach-certs`, or settable with `COCKORACH_CERTS_DIR` environment variable). The individual flags still work in this release but will be removed in the next one. The `--insecure` flag is now *required* for insecure mode. [#14703](https://github.com/cockroachdb/cockroach/pull/14703)
+- The `--alsologtostderr` argument has been renamed to `--logtostderr`, and now takes an optional argument to specify a threshold. [#14841](https://github.com/cockroachdb/cockroach/pull/14841)
+- The `^` operator now means `pow()` instead of bitwise xor (this is consistent with PostgreSQL). [#14882](https://github.com/cockroachdb/cockroach/pull/14882)
```
Comment (mjibson) [inline beta-20170420.md]: May need to add a note: `Any tables with ^ in a DEFAULT or CHECK expression will need to be changed to use # instead.`

File: beta-20170420.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+<div id="os-tabs" class="clearfix">
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170420.darwin-10.9-amd64.tgz"><button id="mac" data-eventcategory="mac-binary-release-notes">Mac</button></a>
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170420.linux-amd64.tgz"><button id="linux" data-eventcategory="linux-binary-release-notes">Linux</button></a>
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170420.windows-6.2-amd64.exe.tgz"><button id="windows" data-eventcategory="windows-binary-release-notes">Windows</button></a>
+</div>
+
+### Backwards-Incompatible Changes
+
+- Security-related command-line arguments have been changed. Instead of individual flags, certificates and keys are now loaded from `--certs-dir` (defaults to `~/.cockroach-certs`, or settable with `COCKORACH_CERTS_DIR` environment variable). The individual flags still work in this release but will be removed in the next one. The `--insecure` flag is now *required* for insecure mode. [#14703](https://github.com/cockroachdb/cockroach/pull/14703)
```
Comment (mberhault) [inline beta-20170420.md]: **bold** (`**bold**`) instead of *italic* (`*italic*`) for `*required* for insecure mode`?

File: beta-20170420.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+- Type errors in `CHECK` constraints are now detected earlier. [#14984](https://github.com/cockroachdb/cockroach/pull/14984)
+- Aggregation functions now work correctly above the windowing level. [#14729](https://github.com/cockroachdb/cockroach/pull/14729)
+- The default precision for inexact decimal operations is now 20 digits (up from 16). [#15001](https://github.com/cockroachdb/cockroach/pull/15001)
+- Bounds are now checked when converting `FLOAT` or `DECIMAL` values to `INTEGER`. [#14967](https://github.com/cockroachdb/cockroach/pull/14967)
+- The `pg_catalog` data for `int2vector` is now consistent with PostgreSQL. [#14989](https://github.com/cockroachdb/cockroach/pull/14989)
+
+### Command-Line Interface Changes
+
+- New arguments `--log-file-max-size` (default 10 MiB) and `--log-dir-max-size` (default 100 MiB) can be used to control log rotation and deletion. `--log-file-verbosity` (default `INFO`) can be used to control the verbosity of the log file. [#14841](https://github.com/cockroachdb/cockroach/pull/14841)
```
Comment (mberhault) [inline beta-20170420.md]: #14703 also reworks all the `cockroach cert` commands. I don't think the details need to be added here, but a mention doesn't hurt.

File: beta-20170420.md (hunk 2 above)
Comment (bdarnell) [inline beta-20170420.md]: Done

File: beta-20170420.md (hunk 1 above)
Comment (bdarnell) [inline beta-20170420.md]: Done

File: beta-20170420.md (hunk 3 above)
Comment (bdarnell) [inline beta-20170420.md]: Done

**Comment by mberhault** [comment]
Still LGTM

**Comment by bdarnell** [comment]
LGTM

---

Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1309#-:-KiCCrEI7MTWtUtKpKQu:bb6ywmx)*

**Pull request merged**